gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -0,0 +1,537 @@
+    }
+
+    /**
+     * Restricts the list to only record sets with this fully qualified domain name.
+     */
+    public static RecordSetListOption dnsName(String dnsName) {
+      return new RecordSetListOption(DnsRpc.Option.NAME, dnsName);
+    }
+
+    /**
+     * Restricts the list to return only record sets of this type. If present, {@link
+     * RecordSetListOption#dnsName(String)} must also be present.
+     */
+    public static RecordSetListOption type(RecordSet.Type type) {
+      return new RecordSetListOption(DnsRpc.Option.DNS_TYPE, type.name());
+    }
+  }
+
+  /**
+   * Class for specifying zone field options.
+   */
+  class ZoneOption extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = -8065564464895945037L;
+
+    ZoneOption(DnsRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the zones's fields to be returned by the RPC call.
+     *
+     * <p>If this option is not provided all zone fields are returned. {@code ZoneOption.fields} can
+     * be used to specify only the fields of interest. Zone ID is always returned, even if not
+     * specified. {@link ZoneField} provides a list of fields that can be used.
+     */
+    public static ZoneOption fields(ZoneField... fields) {
+      return new ZoneOption(DnsRpc.Option.FIELDS, ZoneField.selector(fields));
+    }
+  }
+
+  /**
+   * Class for specifying zone listing options.
+   */
+  class ZoneListOption extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = -2830645032124504717L;
+
+    ZoneListOption(DnsRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the zones's fields to be returned by the RPC call.
+     *
+     * <p>If this option is not provided all zone fields are returned. {@code ZoneOption.fields} can
+     * be used to specify only the fields of interest. Zone ID is always returned, even if not
+     * specified. {@link ZoneField} provides a list of fields that can be used.
+     */
+    public static ZoneListOption fields(ZoneField... fields) {
+      StringBuilder builder = new StringBuilder();
+      builder.append("nextPageToken,managedZones(").append(ZoneField.selector(fields)).append(')');
+      return new ZoneListOption(DnsRpc.Option.FIELDS, builder.toString());
+    }
+
+    /**
+     * Returns an option to specify a page token.
+     *
+     * <p>The page token (returned from a previous call to list) indicates from where listing should
+     * continue.
+     */
+    public static ZoneListOption pageToken(String pageToken) {
+      return new ZoneListOption(DnsRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Restricts the list to only zone with this fully qualified domain name.
+     */
+    public static ZoneListOption dnsName(String dnsName) {
+      StringBuilder builder = new StringBuilder();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://www.codacy.com/app/mziccard/gcloud-java/file/2317493773/issues/source?bid=3123109&fileBranchId=3203698#l305)